engines/misc/src/main/java/org/alfresco/transform/misc/transformers/ImageToPdfTransformer.java

```diff
@@ -0,0 +1,218 @@
+    private static final String DEFAULT_PDF_FORMAT_STRING = "A4";
+    private static final PDRectangle DEFAULT_PDF_FORMAT = PDRectangle.A4;
+
+    @Override
+    public String getTransformerName()
+    {
+        return "imageToPdf";
+    }
+
+    @Override
+    public void transform(
+        String sourceMimetype, String targetMimetype, Map<String, String> transformOptions,
+        File imageFile, File pdfFile, TransformManager transformManager
+    ) throws Exception {
+        try (
+            ImageInputStream imageInputStream = ImageIO.createImageInputStream(imageFile);
+            PDDocument pdfDocument = new PDDocument()
+        ) {
+            final Integer startPage = parseOptionIfPresent(transformOptions, START_PAGE, Integer.class).orElse(null);
+            final Integer endPage = parseOptionIfPresent(transformOptions, END_PAGE, Integer.class).orElse(null);
+            final String pdfFormat = parseOptionIfPresent(transformOptions, PDF_FORMAT, String.class).orElse(DEFAULT_PDF_FORMAT_STRING);
+            verifyOptions(startPage, endPage);
+
+            final ImageReader imageReader = findImageReader(imageInputStream, imageFile.getName(), sourceMimetype);
+            for (int i = 0; i < imageReader.getNumImages(true); i++)
+            {
+                if (startPage != null && i < startPage)
+                {
+                    continue;
+                }
+                if (endPage != null && i > endPage)
+                {
+                    break;
+                }
+
+                scaleAndDrawImage(pdfDocument, imageReader.read(i), pdfFormat);
+            }
+
+            pdfDocument.save(pdfFile);
+        }
+    }
+
+    private ImageReader findImageReader(final ImageInputStream imageInputStream, final String imageName, final String mimetype) throws IOException
+    {
+        final Iterator<ImageReader> imageReaders = ImageIO.getImageReaders(imageInputStream);
+        if (imageReaders == null || !imageReaders.hasNext())
+        {
+            throw new IOException(String.format(INVALID_IMAGE_ERROR_MESSAGE, imageName, mimetype));
+        }
+        final ImageReader imageReader = imageReaders.next();
+        imageReader.setInput(imageInputStream);
+
+        return imageReader;
+    }
+
+    private void scaleAndDrawImage(final PDDocument pdfDocument, final BufferedImage bufferedImage, final String pdfFormat) throws IOException
+    {
+        final PDPage pdfPage = new PDPage(resolvePdfFormat(pdfFormat));
+        pdfDocument.addPage(pdfPage);
+        final PDImageXObject image = LosslessFactory.createFromImage(pdfDocument, bufferedImage);
+        try (PDPageContentStream pdfPageContent = new PDPageContentStream(pdfDocument, pdfPage))
+        {
+            final PDRectangle pageSize = pdfPage.getMediaBox();
+            final float withRatio = pageSize.getWidth() / image.getWidth();
+            final float heightRatio = pageSize.getHeight() / image.getHeight();
```

Comment:
Do we need to handle edge cases around dimensions being zero?

Reply:
Thanks! Fixed.